pkgs/os-specific/darwin/apple-sdk-11.0/cf-setup-hook.sh

```diff
@@ -1,6 +1,14 @@
 forceLinkCoreFoundationFramework() {
-  NIX_CFLAGS_COMPILE="-F@out@/Library/Frameworks${NIX_CFLAGS_COMPILE:+ }${NIX_CFLAGS_COMPILE-}"
-  NIX_LDFLAGS+=" @out@/Library/Frameworks/CoreFoundation.framework/CoreFoundation.tbd"
+  local role_post
+  getHostRole
+
+  local cflags_compile_role_var="NIX_CFLAGS_COMPILE${role_post}"
+  local cflags_compile_role="${!cflags_compile_role_var-}"
+  export NIX_CFLAGS_COMPILE${role_post}="-F@out@/Library/Frameworks${cflags_compile_role:+ $cflags_compile_role}"
+
+  local ldflags_role_var="NIX_LDFLAGS${role_post}"
+  local ldflags_role="${!ldflags_role_var-}"
+  export NIX_LDFLAGS${role_post}+="${ldflags_role:+ }@out@/Library/Frameworks/CoreFoundation.framework/CoreFoundation.tbd"
```

Comment:
Isn't this appending the entire value of `NIX_LDFLAGS${role_post}` again? Do we want to be outright setting the variable here or keep it to just appending the CoreFoundation.tbd path?

Reply:
`${ldflags_role:+ }` prepends a space if the variable is not empty. It doesn't include the content of the variable. It's probably not necessary and we could always include an extra space to simplify it. I just tried to keep the functionality as close as possible to the original.

Reply:
Totally right, misread the expansion entirely. That particular pattern is very confusing in a Github code block that wraps at whitespace >.<